test/performance/ibf/bit_vector_serialisation_benchmark.cpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: 2006-2023, Knut Reinert & Freie Universität Berlin
+// SPDX-FileCopyrightText: 2016-2023, Knut Reinert & MPI für molekulare Genetik
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <benchmark/benchmark.h> // for Benchmark, State, BENCHMARK_CAPTURE, DoNotOptimize, BENCHMARK_MAIN
+
+#include <algorithm> // for __generate_fn, generate
+#include <cinttypes> // for int32_t, uint8_t
+#include <cstddef>   // for size_t
+#include <filesystem>
+#include <fstream>
+#include <memory>      // for allocator
+#include <random>      // for uniform_int_distribution, mt19937_64
+#include <type_traits> // for invoke_result_t
+#include <utility>     // for move, pair
+
+#include <hibf/misc/bit_vector.hpp>     // for bit_vector
+#include <hibf/test/sandboxed_path.hpp> // for sandboxed_path, operator/
+#include <hibf/test/tmp_directory.hpp>  // for tmp_directory
+
+#include <cereal/archives/binary.hpp> // for BinaryInputArchive, BinaryOutputArchive
+
+seqan::hibf::bit_vector generate_bit_vector(size_t const size_in_bits, size_t const seed = 0u)
+{
+    std::mt19937_64 engine{seed};
+    std::uniform_int_distribution<uint8_t> dist{0u, 1u};
+
+    auto gen = [&dist, &engine]()
+    {
+        return dist(engine);
+    };
+    seqan::hibf::bit_vector vec(size_in_bits);
+    std::ranges::generate(vec, gen);
```

Comment:
```suggestion
    seqan::hibf::bit_vector vec(size_in_bits);
    std::ranges::generate(vec, dist(engine));
```
does this not accept a temporary?